### ISSUE TITLE
Wrapping up this repo

### ISSUE THREAD

**Issue by amanjeev** [CanDIG/candig_tyk#4]
This repo can be archived as its source is mainly in candig_compose and the docs should be in the runbooks repo.

- [ ] Create a new repo for Gitbook (or something similar)
- [ ] Move the docs in this repo to that new repo (https://github.com/CanDIG/candig_tyk/issues/9)
- [x] Merge `authn_decoupling` to `master`
- [x] All sites are on Tyk Authn architecture
- [x] The [middlewares and virtual endpoint source is in candig_compose repo](https://github.com/CanDIG/candig_compose/tree/master/template/config.tpl/tyk/confs)
- [ ] Archive this repo

cc/ @haoyuanli , @poquirion

**Comment by jimmyhli** [comment]
candig_compose is for Docker deployment, right? I wonder if it will be unclear to sites with non-docker deployment

**Comment by amanjeev** [comment]
@haoyuanli : excellent point!

My $0.02 is to reduce duplication. I know I will eventually forget to update the source in one place if the same file exists in multiple places. 

If we keep this separate repo for Tyk config, then I believe it will be better to incorporate this repo in `candig_compose` in some(1) way rather than duplicating the effort. 

---
1. I do not know that way yet. The easiest is git submodules but that solution is a bit cumbersome historically (maybe things have changed).

**Comment by ljdursi** [comment]
+1 to multiple deployments - at some point we'd like to have two reference deployments; +1 too to avoiding duplication.

The duplication thing will come up in other contexts too as we move to multiple services.  Git submodules, packages, libraries... - it'd be nice to have one usual way of doing it but the right solution may depend on the type of thing being encapsulated and re-used.

**Comment by jimmyhli** [comment]
> @haoyuanli : excellent point!
> 
> My $0.02 is to reduce duplication. I know I will eventually forget to update the source in one place if the same file exists in multiple places.
> 
> If we keep this separate repo for Tyk config, then I believe it will be better to incorporate this repo in `candig_compose` in some(1) way rather than duplicating the effort.
> 
> 1. I do not know that way yet. The easiest is git submodules but that solution is a bit cumbersome historically (maybe things have changed).

Yes, and i agree it's better to have a single source of truth here (e.g., for the middleware). Let me check a few things

**Comment by amanjeev** [comment]
Thinking out loud - 

Maybe we need to have a single source of config generation that can be used in these two cases of CanDIG suite's

1. Standalone deployment
2. Dockerized deployment

We could restructure the common parts. What do you think?

**Comment by amanjeev** [comment]
@haoyuanli @poquirion : 

To finalize this repo, I have created PR to merge the authentication decoupling code to `master` branch. https://github.com/CanDIG/candig_tyk/pull/5

This merge has been long overdue and is mainly to reduce confusion between what is in the master (which is old) and what is the middleware for authentication lifting to Tyk.

This repo contains valuable configs for Tyk middleware etc. What do you think about using the code in this and moving it to the common repo for deployment? cc/ @SamarthVP and @shaikh-rashid

**Comment by amanjeev** [comment]
One step closer to archiving this. https://github.com/CanDIG/candig_tyk/issues/9
